### DISCUSSION TITLE
Should we use a 4096-word list

### DISCUSSION THREAD

**Discussion by kklash** [kklash/mnemonikey#21] [Specification Improvements]
One option I've considered is breaking compatibility with BIP39's wordlists, and creating our own custom 4096-word list, each word encoding 12 bits of information. This would allow us a more densely-packed encoding scheme, so we can fit more data into fewer words. 

I've been hesitant to do this, because if we break compatibility with BIP39, we'd be adding yet another wordlist to the heap [(see relevant XKCD)](https://xkcd.com/927/). We would no longer have the option of easily adding internationalization support via [the existing BIP39 wordlists already available for other languages.](https://github.com/bitcoin/bips/blob/master/bip-0039/bip-0039-wordlists.md)

Furthermore, designing a wordlist is tedious work, and I have a day job :joy: 

**Does anyone know of any pre-existing well-designed wordlists of length 4096 which might be appropriate?** Ideally we want something which fits the common requirements of a BIP39-style wordlist:

- Avoid homophones (`arc` vs `ark`, `beet` vs `beat`)
- Avoid similar related words (`women` vs `woman`)
- Avoid plurals (`cactus` vs `cacti`)
- Each word should have a unique prefix 
  - From BIP39: `the wordlist is created in such a way that it's enough to type the first four letters to unambiguously identify the word`

I already have a prototype wordlist which I spent a day or two constructing last month, built from SLIP39, BIP39, and diceware wordlists. It'd be better if we could find something which already exists to reference.

**Comment by RyanZim** [comment]
> We would no longer have the option of easily adding internationalization support via [the existing BIP39 wordlists already available for other languages.](https://github.com/bitcoin/bips/blob/master/bip-0039/bip-0039-wordlists.md)

Internationalized wordlists increase complexity and decrease portability; it's not at all obvious that internationalization is a goal worthy of pursuing. See https://github.com/bitcoin/bips/wiki/Comments:BIP-0039#electrum-criticism-of-bip39 for reference.

**Reply by kklash** [reply]
Fair point!

**Comment by kklash** [comment]
I've got a work in progress coming for this. Turns out it's a bit easier than i thought. Still a bit tedious but i think it will be worth it for the extra encoding density.

**Comment by kklash** [comment]
Closing as resolved by #26
